### ISSUE TITLE
about hardware cursor processor

### ISSUE THREAD

**Issue by Sampsonksw** [ZipCPU/vgasim#5]
Mr.Gisselquist

I am very interested in your video interface project. I would like to know if there is a hardware cursor processor in the vgasim

**Comment by ZipCPU** [comment]
> I would like to know if there is a hardware cursor processor in the vgasim

Not in vgasim at present, no.

I do have a hardware cursor processor that I built around 3 yrs ago, but 1) I never released it, and 2) it's not really up to my coding standards today, 3) the sprites I used for my hardware cursors left some ... artistic qualities to be desired.   :D  I'm tempted to rebuild it as part of a sprite processor operating on a well defined video stream--something like an AXI stream video format or some such.  It's not that hard to do--only perhaps 160 lines of code or so.  I might even build one over a weekend sometime.

I'd imagine the ideal processor would contain:

1. A WB (or other) interface for adjusting the (24b color + alpha) pixels of the sprite, as well as it's location on the screen

2. Stream inputs and outputs

3. I'm not quite sure how I'd handle alpha blending today.  The original method was to use a 1-bit mask, which would be easy enough to duplicate.  The "right" way would probably be to use three multiplies, but multiplies are expensive so ... I'll have to think about that a bit.

Dan
